base/pmap.jl

```diff
@@ -64,9 +58,13 @@ Split a collection into at least `min_batch_count` batches.
 Equivalent to `partition(c, max_batch_size)` when `length(c) >> max_batch_size`.
 """
 function batchsplit(c; min_batch_count=1, max_batch_size=100)
+    if min_batch_count < 1
+        throw(ArgumentError("min_batch_count must be > 0, got $min_batch_count"))
+    end
 
-    @assert min_batch_count > 0
-    @assert max_batch_size > 1
+    if max_batch_size < 1
```

Comment:
I don't mind if you change this. I'll try to explain my point of view as best I can and leave it at that...
Disclaimer: My precondition design choices are influenced by being beaten by the Eiffel/DBC stick at a young age.

My feeling is that preconditions should protect against:
 - input that is invalid -- because that will stop the implementation from crashing or computing an incorrect result.
 - input that is not sane/useful -- because alerting the user early in the development process that they are doing something that makes no sense might prompt them to understand things better and write better code

I also feel that when in doubt it is better to make preconditions tighter rather than looser because:
 - it reduces the number of cases that need to be covered in testing
 - it reduces the metal load on code reviewers (e.g. considering, "will it still work in the corner case of max=1?")
 - if someone one day finds a good reason to make the precondition looser, that's easy to do without breaking anyone else's code (making a precondition tighter can break code).
 - it provides a clear indication to the user, that the author of the function intended it to be used in a certain way.